### PR TITLE
docs: limits on integer and number option types

### PR DESCRIPTION
**Pull request by JustaSqu1d**
## Summary

Many people seem to use `SlashCommandOptionType.integer` or similar for IDs. This change serves to clarify its limitations.

The superscripts seem to have extra spacing because of monospaces, but this looks fine-ish on the documentation.

Looking for feedback about whether the ID notes for `SlashCommandOptionType.number` are necessary or can be removed.

I'm not sure how to remove ambiguity with this change and with `Option.max_value` and `Option.min_value` since those can be used to change the range of accepted values.

image

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by plun1331** [comment]
Should probably just replace the docs for that instead of making a note

File: docs/api/enums.rst [hunk 1]
```diff
@@ -29,7 +29,12 @@ of :class:`enum.Enum`.
         A string.
     .. attribute:: integer
 
-        An integer.
+        An integer between -2⁵³ and 2⁵³.
+
+        .. note::
+
+            IDs, such as 881224361015672863, are often too for this input type.
```
Comment (Dorukyum) [inline docs/api/enums.rst:36]:
```suggestion
            IDs, such as 881224361015672863, are often too big for this input type.
```
or any other adjective

File: docs/api/enums.rst [hunk 2]
```diff
@@ -47,7 +52,11 @@ of :class:`enum.Enum`.
         A mentionable (user or role).
     .. attribute:: number
 
-        A floating number.
+        A floating-point number between -2⁵³ and 2⁵³.
+
+        .. note::
+
+            IDs, such as 881224361015672863, are often too for this input type.
```
Comment (Dorukyum) [inline docs/api/enums.rst:59]:
```suggestion
            IDs, such as 881224361015672863, are often too big for this input type.
```

Comment (Dorukyum) [inline docs/api/enums.rst:59]:
```suggestion
            IDs, such as 881224361015672863, are often too big for this input type.
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Comment by Dorukyum** [comment]
Does specifying the limits as exponents of 2 actually help with anything?

**Comment by JustaSqu1d** [comment]
> Does specifying the limits as exponents of 2 actually help with anything?

Would you rather 9.0071992547*10^15?

**Comment by Dorukyum** [comment]
> > Does specifying the limits as exponents of 2 actually help with anything?
> 
> 
> 
> Would you rather have 9.0071992547*10^15? Or am I misunderstanding the question?

You did get that right and I did mean that, but now I think we should specify both, like "9007199254740992 (2^53)".

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**